### PR TITLE
Mimic: hold the target of each gripper transition before commanding it

### PR DESCRIPTION
**Pull request by 2047767028-lang**
# Description

MimicGen replays a source subtask **by frame index**: the gripper command is issued on the frame it had in the source demonstration, whatever the generated arm is doing at that moment. The human who recorded the source did something the replay drops: they stopped the arm before closing or opening the gripper. Over the 10 shipped Franka stack-cube source demos, `|target_eef_pose - eef_pose|` is **4.86 cm median on ordinary frames and 0.00 cm on the gripper-command frames** (97% of them under 0.3 cm). The generated arm, driven with the default `action_noise=0.03` from a different joint configuration, is still moving when that frame arrives: 1.12 cm from the cube at closure against the human's 0.56 cm, and the cube is released 3.7 cm (p90) off the cube below, past the 2.34 cm tipping threshold. The placement and the second grasp pay for it.

This PR adds one `SubTaskConfig` option, **off by default**, that gives the generated arm the pause the operator had:

```python
num_settle_steps_before_gripper: int = 0
```

When set, `insert_settle_frames_before_gripper()` inserts, before every gripper transition of the transformed subtask segment, that many copies of the transition's target pose with the pre-transition gripper action and **zero action noise**, so the arm settles onto the target before the gripper acts. Nothing else in the pipeline changes; with the default 0 the trajectory is byte-identical to today's.

`num_fixed_steps` already exists and is also a noise-free hold, but `WaypointTrajectory.merge()` places it at the **start of the subtask** (the subtask's first pose), a whole subtask and 10–20 offset frames away from the gripper event; every shipped config sets it to 0. NVlabs/mimicgen has the same `merge`, so the gap is inherited from the original design rather than introduced by the port.

## Evidence

All runs: stock `Isaac-Stack-Cube-Franka-IK-Rel-Mimic-v0`, the shipped annotated source dataset, `num_envs=10`, 300 attempts per run, `generation_guarantee=False`, five scene draws (per-episode reseeding so the settings of a draw see the same scenes). ± is the standard error of the mean paired delta over the five draws; a single run's binomial SE is about 2.8 pp. Measured on Isaac Lab 2.3.2 + this patch (the change applies unmodified to both).

| setting | d1 | d2 | d3 | d4 | d5 | mean | Δ vs stock | episode frames |
|---|---|---|---|---|---|---|---|---|
| stock (0 steps) | 30.3 | 38.7 | 34.7 | 36.7 | 36.7 | 35.4 | — | 230 |
| hold 5 | 45.0 | 51.0 | 46.3 | 48.0 | 49.3 | 47.9 | +12.5 ± 0.6 | 250 |
| hold 10 | 48.7 | 53.0 | 51.0 | 50.0 | 52.3 | 51.0 | +15.6 ± 0.9 | 270 |
| **hold 20** | 54.7 | 53.3 | 54.0 | 51.0 | 56.7 | **53.9** | **+18.5 ± 1.9** | 310 |
| hold 30 | 54.3 | 56.7 | 56.7 | 55.3 | 56.3 | 55.9 | +20.5 ± 1.1 | 350 |
| hold 40 | 53.3 | 53.7 | 55.7 | 53.3 | 56.0 | 54.4 | +19.0 ± 1.4 | 390 |
| hold 60 | 50.0 | 54.0 | 56.3 | 54.3 | 58.3 | 54.6 | +19.2 ± 1.2 | 470 |

Mechanism (cm, median / p90, pooled over draws): gripper-to-cube distance at the first closure 1.17 / 2.10 → 0.83 / 1.73 (source demos 0.56); cube-to-cube xy offset at the first release 1.25 / 3.66 → 0.87 / 1.65 (source demos 0.90, tipping threshold 2.34). The curve is flat from 20 steps on: the arm controller is a first-order tracker with a ~10-frame time constant, so a few time constants is the rule and there is no penalty for overshooting it except episode length.

Ablations (same five draws):

| variant | mean | Δ vs stock | what it says |
|---|---|---|---|
| stock `num_fixed_steps=20` (hold at subtask start, same +80 frames) | 45.7 | +10.3 ± 1.8 | position matters: the arm still arrives unsettled (closure 1.16 cm) |
| hold 20 with the action noise kept on the held frames | 39.3 | +3.9 ± 1.6 | noise-free is most of the effect (closure 1.26 cm) |
| hold 20 before closing transitions only | 44.8 | +9.4 ± 1.3 | both transition types matter, roughly additively |
| hold 20 before opening transitions only | 42.9 | +7.5 ± 0.9 | |
| convergence-gated hold (repeat until within 0.3 cm, ≤40) | 54.5 | +19.1 ± 1.3 | no better than the fixed hold; not included to keep the change small |

Boundary: on `Isaac-Stack-Cube-Franka-IK-Abs-Mimic-v0` (absolute-pose IK, `action_noise=0.01`) the arm already arrives settled (stock release p90 1.6 cm, where IK-Rel only gets *with* the hold) and the option is neutral: 84.0% → 83.1%, −0.9 ± 0.6 pp. The option helps when the generated arm reaches the gripper frames unsettled, i.e. a lagging controller driven with sizeable action noise; the source demos' residual at the gripper frames (≈0) against a stock run's release offset is the tell.

## Tests

`source/isaaclab_mimic/test/test_settle_before_gripper.py` covers the insertion as pure tensor logic: zero steps and no-transition inputs returned unchanged, a single frame, copies inserted before every transition with the pre-transition gripper action and zero noise (1- and 2-DoF grippers), per-frame noise tensors zeroed on the held frames only, a transition on the last frame, and the held frames surviving the interpolate-merge-pop that `merge_eef_subtask_trajectory` performs.

The option assumes discrete open/close gripper commands (a transition is any change of the gripper action between consecutive frames); the docstring says to leave it at 0 for continuous hand-joint targets, where nearly every frame differs.

## Type of change

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read and understood the [contribution guidelines](https://isaac-sim.github.io/IsaacLab/main/source/refs/contributing.html)
- [x] I have run the `pre-commit` checks on the changed files (ruff, codespell, and the repository hooks)
- [x] I have made corresponding changes to the documentation (config docstrings)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added a changelog fragment under `source/<pkg>/changelog.d/` for every touched package
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

**Comment by 2047767028-lang** [comment]
Follow-up in 4ed8af0 for the three automated findings:

**Boundary transitions skip settling** (https://github.com/isaac-sim/IsaacLab/pull/7734#discussion_r3979756236) — correct. The insertion only compared consecutive frames inside the segment, so a segment whose first frame already carries the new gripper action (the transition sits between the previous segment's last source frame and this one's first) was executed through the interpolation segment with the new action and no hold. `generate_eef_subtask_trajectory` now passes the source gripper action just before the segment (`previous_gripper_action`); a differing first frame is held at its target with the previous action, and because that hold leads the segment, the frame `WaypointTrajectory.merge` pops as the interpolation target carries the pre-transition action too. Three new tests cover it, including the merged sequence (`[-1] * 8 + [1] * 3`). None of the shipped configs hit this case today (their term signals fire ~10 frames after the transition, so it sits inside the segment), but the option should hold every transition, not most.

**Settle frames desynchronize effectors** (https://github.com/isaac-sim/IsaacLab/pull/7734#discussion_r3979756250) — correct, and not something this PR should paper over: `synchronous_steps` is a count of source frames, and the hold lengthens each end effector's segment by its own number of transitions. Rather than making the coordination logic hold-aware here, `DataGenerator.__init__` now refuses the combination (`check_settle_hold_compatible` raises `ValueError` when `num_settle_steps_before_gripper` is set on a subtask under a `COORDINATION` constraint; sequential constraints are unaffected). Documented on the field and in the changelog fragment; tested. No shipped task combines a coordination constraint with a discrete gripper, and the tasks that do coordinate (GR1T2 / G1) use continuous hand-joint targets, for which the docstring already says to leave the option at 0.

**Scalar noise tensor ignores pose device** — the scalar branch mirrored `WaypointSequence.from_poses`, which also builds its per-frame noise on the CPU (`action_noise * torch.ones(...)`) and works because the consumer multiplies a 0-dim tensor with `torch.randn_like(pose_action)`. Still, there is no reason for the helper to return mixed devices, so both branches now allocate on `poses.device`; a test asserts it (on CUDA when available).

Both bots' other points stand as described in the PR text; nothing else changed.
